File: 2004b/main.py
def I():
    return input()


def II():
    return int(input())


def MII():
    return map(int, input().split())


def LI():
    return list(input().split())


def LII():
    return list(map(int, input().split()))

# alpha = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']

a = II()
for _ in range(a):
    x, y = MII()
    m, n = MII()

    if x > n or m > y:
        print(1)
        continue

    ll, rr = max(x, m), min(y, n)
    xl, xr = min(x, m), max(y, n)

    ans = rr - ll
    if xl < ll :
        ans += 1
    if xr > rr:
        ans += 1

    print(ans)

File: 2004b/main.go
package main

import (
	"bufio"
	. "fmt"
	"io"
	"os"
)

func solve(in *bufio.Reader, out *bufio.Writer) {
	var x, y, m, n int
	Fscan(in, &x, &y, &m, &n)
	if m > y || x > n {
		Fprintln(out, 1)
		return
	}

	ll, rr := max(x, m), min(y, n)
	xl, xr := min(x, m), max(y, n)

	ans := rr - ll
	if xl < ll {
		ans += 1
	}
	if rr < xr {
		ans += 1
	}
	Fprintln(out, ans)
}

func run(r io.Reader, w io.Writer) {
	in := bufio.NewReader(r)
	out := bufio.NewWriter(w)
	defer out.Flush()
	var a int
	Fscan(in, &a)
	for i := 0; i < a; i++ {
		solve(in, out)
	}

}

func main() {
	run(os.Stdin, os.Stdout)
}
